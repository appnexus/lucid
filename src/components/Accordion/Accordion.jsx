import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { lucidClassNames } from '../../util/style-helpers';
import {
	createClass as createReactClass,
	findTypes,
	omitProps,
} from '../../util/component-types';
import { buildHybridComponent } from '../../util/state-management';

import {
	ExpanderPanelDumb as ExpanderPanel,
} from '../ExpanderPanel/ExpanderPanel';

import * as reducers from '../Accordion/Accordion.reducers';

const cx = lucidClassNames.bind('&-Accordion');

const { any, func, object, number, string } = PropTypes;

/**
* {"categories": ["layout"], "madeFrom": ["ExpanderPanel"]}
*
* This is a container that renders panels and controls its expansion/retraction.
*/
const Accordion = createReactClass({
	displayName: 'Accordion',

	components: {
		Item: ExpanderPanel,
		Header: ExpanderPanel.Header,
	},

	reducers,

	propTypes: {
		/**
		* Appended to the component-specific class names set on the root
		* element.
		*/
		className: string,

		/**
		* Indicates which item is expanded.
		*/
		selectedIndex: number,

		/**
		* Called when the user clicks on the component's header of an item.
		*
		* Signature: `(itemIndex, { event, props }) => {}`
		*/
		onSelect: func,

		/**
		* Passed through to the root element.
		*/
		style: object,

		/**
		 * prop alternative to Header child component
		 * passed through to the underlying ExpanderPanel
		 */
		Header: any,
	},

	getDefaultProps() {
		return {
			onSelect: _.noop,
		};
	},

	handleToggle(isExpanded, index, event) {
		const selectedIndex = isExpanded ? index : null;

		this.props.onSelect(selectedIndex, {
			event,
			props: this.props,
		});
	},

	render() {
		const { style, className, selectedIndex, ...passThroughs } = this.props;

		const itemChildProps = _.map(
			findTypes(this.props, Accordion.Item),
			'props'
		);

		return (
			<div
				{...omitProps(passThroughs, Accordion)}
				className={cx('&', className)}
				style={style}
			>
				{_.map(itemChildProps, (itemChildProp, index) => {
					return (
						<ExpanderPanel
							key={index}
							{...itemChildProp}
							className={cx('&-Item', itemChildProp.className)}
							onToggle={(isExpanded, { event }) =>
								this.handleToggle(isExpanded, index, event)}
							isExpanded={!itemChildProp.isDisabled && selectedIndex === index}
						/>
					);
				})}
			</div>
		);
	},
});

export default buildHybridComponent(Accordion);
export { Accordion as AccordionDumb };
