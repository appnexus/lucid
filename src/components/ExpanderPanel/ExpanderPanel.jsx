import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { lucidClassNames } from '../../util/style-helpers';
import {
	createClass as createReactClass,
	getFirst,
	omitProps,
} from '../../util/component-types';
import { buildHybridComponent } from '../../util/state-management';

import ChevronIcon from '../Icon/ChevronIcon/ChevronIcon';
import Collapsible from '../Collapsible/Collapsible';
import Panel from '../Panel/Panel';

import * as reducers from '../Expander/Expander.reducers';

const cx = lucidClassNames.bind('&-ExpanderPanel');

const { any, bool, func, node, object, string } = PropTypes;

/**
 * {"categories": ["layout"], "madeFrom": ["ChevronIcon", "Expander", "Panel"]}
 *
 * This is a container that provides a toggle that controls when the content is
 * shown.
 */
const ExpanderPanel = createReactClass({
	displayName: 'ExpanderPanel',

	components: {
		/**
		 * Renders a `<span>` of content next to the `ChevronIcon` in the `Panel.Header`
		 */
		Header: createReactClass({
			displayName: 'ExpanderPanel.Header',
			propName: 'Header',
			propTypes: {
				/**
				 * Used to identify the purpose of this switch to the user -- can be
				 * any renderable content.
				 */
				children: node,
			},
		}),
	},

	reducers,

	propTypes: {
		/**
		 * Expandable content.
		 */
		children: node,

		/**
		 * Appended to the component-specific class names set on the root
		 * element.
		 */
		className: string,

		/**
		 * Indicates that the component is in the "expanded" state when true
		 * and in the "unexpanded" state when false.
		 */
		isExpanded: bool,

		/**
		 * Indicates that the component is in the "disabled" state when true
		 * and in the "enabled" state when false.
		 */
		isDisabled: bool,

		/**
		 * Controls the presence of padding on the inner content.
		 */
		hasPadding: bool,

		/**
		 * Called when the user clicks on the component's header.
		 *
		 * Signature: `(isExpanded, { event, props }) => {}`
		 */
		onToggle: func,

		/**
		 * Passed through to the root element.
		 */
		style: object,

		/**
		 * prop alternative to Header child component
		 * passed through to the underlying ExpanderPanel
		 */
		Header: any,
	},

	getDefaultProps() {
		return {
			isExpanded: false,
			onToggle: _.noop,
			hasPadding: true,
		};
	},

	handleToggle(event) {
		if (!this.props.isDisabled) {
			this.props.onToggle(!this.props.isExpanded, {
				event,
				props: this.props,
			});
		}
	},

	render() {
		const {
			children,
			className,
			isExpanded,
			isDisabled,
			hasPadding,
			style,
			...passThroughs
		} = this.props;

		const headerChildProps = _.get(
			getFirst(this.props, ExpanderPanel.Header),
			'props'
		);

		return (
			<Panel
				{...omitProps(passThroughs, ExpanderPanel, [], false)}
				className={cx(
					'&',
					{
						'&-is-collapsed': !isExpanded,
						'&-is-disabled': isDisabled,
					},
					className
				)}
				style={style}
			>
				<Panel.Header className={cx('&-header')} onClick={this.handleToggle}>
					<span className={cx('&-icon')}>
						<ChevronIcon direction={isExpanded ? 'up' : 'down'} />
					</span>

					<span {...headerChildProps} />
				</Panel.Header>

				<Collapsible
					isExpanded={isExpanded}
					className={cx('&-content', {
						'&-content-is-expanded': isExpanded,
					})}
				>
					<div
						className={cx('&-content-inner', {
							'&-content-inner-has-padding': hasPadding,
						})}
					>
						{children}
					</div>
				</Collapsible>
			</Panel>
		);
	},
});

export default buildHybridComponent(ExpanderPanel);
export { ExpanderPanel as ExpanderPanelDumb };
