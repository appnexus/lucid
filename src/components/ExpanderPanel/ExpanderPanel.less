@import (reference) '../../styles/variables.less';
@import (reference) '../../styles/mixins.less';

.lucid-ExpanderPanel {
	&-icon {
		.box-sizing();
		width: @size-expander-button;
		height: @size-expander-button;
		display: flex;
		align-items: center;
		justify-content: center;
		margin-right: @size-S;
	}

	&-header {
		font-size: @size-font-XL;
		line-height: @size-font-XL;
		font-weight: bold;
		cursor: pointer;
		display: flex;
		align-items: center;

		&:hover {
			> .lucid-ExpanderPanel-icon {
				.gradient-animation(@featured-color-default-gradientStartColor, @featured-color-default-gradientEndColor);
				border: solid 1px @featured-color-default-borderColor;
				border-radius: @size-borderRadius;
				outline: none;

				.lucid-ChevronIcon {
					fill: @color-darkGray;
				}
			}
		}
	}

	&-content-inner {
		border-top: @border-lightBorder;

		&-has-padding {
			padding: @size-standard;
		}
	}

	> .lucid-Panel-content.lucid-Panel-content { // increased specificity
		padding: 0;
	}

	> .lucid-Panel-Header {
		border-bottom: none;
	}

	&-is-disabled {
		.opacity();

		&:after {
			content: '';
			position: absolute;
			top: 0;
			left: 0;
			right: 0;
			bottom: 0;
			color: @color-disabledText;
			.no-cursor();
		}
	}

}
