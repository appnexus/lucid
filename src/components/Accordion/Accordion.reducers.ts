import { IAccordionState } from './Accordion';

export function onSelect(
	state: IAccordionState,
	selectedIndex: number
): IAccordionState {
	return {
		...state,
		selectedIndex,
	};
}
