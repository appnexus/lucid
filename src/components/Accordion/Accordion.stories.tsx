import React, { useState } from 'react';
import { Meta } from '@storybook/react';
import Accordion, { AccordionDumb } from './Accordion';

export default {
	title: 'Layout/Accordion',
	component: AccordionDumb,
	subcomponents: { 'Accordion.Item': Accordion.Item },
	parameters: {
		notes: AccordionDumb.peek.description,
		docs: {
			inlineStories: false,
		},
	},
} as Meta;

export const Default = () => (
	<Accordion>
		<Accordion.Item Header='Item 1'>Item 1 Content</Accordion.Item>
	</Accordion>
);

export const Basic = () => (
	<Accordion>
		<Accordion.Item Header='Peter Venkman'>
			Peter Venkman, Ph.D. is a fictional character from the Ghostbusters
			franchise. He appears in the films Ghostbusters and Ghostbusters II and in
			the animated television series The Real Ghostbusters. In both live action
			films, he was portrayed by Bill Murray, and was voiced in the animated
			series first by Lorenzo Music and then by Dave Coulier. He is a
			parapsychologist and the leader of the Ghostbusters.
		</Accordion.Item>
		<Accordion.Item Header='Ray Stantz'>
			Raymond Ray Stantz, Ph.D. is a fictional character from the Ghostbusters
			franchise. He appears in the films Ghostbusters, Ghostbusters II, Casper,
			and the animated television series The Real Ghostbusters. He was portrayed
			by Dan Aykroyd in both live action films, and voiced by Frank Welker in
			the animated series. He is a member of the Ghostbusters and one of the
			three doctors of parapsychology, along with Dr. Peter Venkman and Dr. Egon
			Spengler.
		</Accordion.Item>
		<Accordion.Item Header='Egon Spengler'>
			Egon Spengler, Ph.D. is a fictional character from the Ghostbusters
			franchise. He appears in the films Ghostbusters and Ghostbusters II, in
			the animated television series The Real Ghostbusters, and later in Extreme
			Ghostbusters. Spengler was portrayed by Harold Ramis in the films and
			voiced by him in Ghostbusters: The Video Game, and voiced by Maurice
			LaMarche in the cartoon series. He is a member of the Ghostbusters and one
			of the three doctors of parapsychology, along with Dr. Peter Venkman and
			Dr. Raymond Stantz.
		</Accordion.Item>
	</Accordion>
);

export const InitialIndexSet = () => {
	const [index, setIndex] = useState(2);

	const handleChange = (index: number) => {
		setIndex(index);
	};

	return (
		<Accordion onSelect={handleChange} selectedIndex={index}>
			<Accordion.Item Header='Peter Venkman'>
				Peter Venkman, Ph.D. is a fictional character from the Ghostbusters
				franchise. He appears in the films Ghostbusters and Ghostbusters II and
				in the animated television series The Real Ghostbusters. In both live
				action films, he was portrayed by Bill Murray, and was voiced in the
				animated series first by Lorenzo Music and then by Dave Coulier. He is a
				parapsychologist and the leader of the Ghostbusters.
			</Accordion.Item>
			<Accordion.Item Header='Ray Stantz'>
				Raymond Ray Stantz, Ph.D. is a fictional character from the Ghostbusters
				franchise. He appears in the films Ghostbusters, Ghostbusters II,
				Casper, and the animated television series The Real Ghostbusters. He was
				portrayed by Dan Aykroyd in both live action films, and voiced by Frank
				Welker in the animated series. He is a member of the Ghostbusters and
				one of the three doctors of parapsychology, along with Dr. Peter Venkman
				and Dr. Egon Spengler.
			</Accordion.Item>
			<Accordion.Item Header='Egon Spengler'>
				Egon Spengler, Ph.D. is a fictional character from the Ghostbusters
				franchise. He appears in the films Ghostbusters and Ghostbusters II, in
				the animated television series The Real Ghostbusters, and later in
				Extreme Ghostbusters. Spengler was portrayed by Harold Ramis in the
				films and voiced by him in Ghostbusters: The Video Game, and voiced by
				Maurice LaMarche in the cartoon series. He is a member of the
				Ghostbusters and one of the three doctors of parapsychology, along with
				Dr. Peter Venkman and Dr. Raymond Stantz.
			</Accordion.Item>
		</Accordion>
	);
};
