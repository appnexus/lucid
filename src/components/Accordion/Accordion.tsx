import _ from 'lodash';
import React from 'react';
import PropTypes from 'react-peek/prop-types';
import { lucidClassNames } from '../../util/style-helpers';
import { buildModernHybridComponent } from '../../util/state-management';
import { ExpanderPanelDumb as ExpanderPanel } from '../ExpanderPanel/ExpanderPanel';
import {
	findTypes,
	omitProps,
	StandardProps,
} from '../../util/component-types';
import * as reducers from '../Accordion/Accordion.reducers';

const cx = lucidClassNames.bind('&-Accordion');

const { func, object, number, string } = PropTypes;
interface IAccordionProps extends StandardProps {
	/**
	 * Indicates which item is expanded
	 * */
	selectedIndex?: number | null;

	/**
	 * Called when the user clicks on the component's header of an item.
	 * */
	onSelect: (
		selectedIndex: number | null,
		{ event, props }: { event: React.MouseEvent; props: IAccordionProps }
	) => void;
}

export interface IAccordionState {
	selectedIndex: number | null;
}

const defaultProps = {
	onSelect: _.noop,
};

const Accordion = (props: IAccordionProps): React.ReactElement => {
	const { style, className, selectedIndex, ...passThroughs } = props;

	const itemChildProps = _.map(findTypes(props, Accordion.Item), 'props');

	const handleToggle = (
		isExpanded: boolean,
		index: number,
		event: React.MouseEvent
	) => {
		const selectedIndex = isExpanded ? index : null;

		props.onSelect(selectedIndex, {
			event,
			props,
		});
	};

	return (
		<div
			{...omitProps(passThroughs, undefined, _.keys(Accordion.propTypes))}
			className={cx('&', className)}
			style={style}
		>
			{_.map(itemChildProps, (itemChildProp, index: number) => {
				return (
					<ExpanderPanel
						key={index}
						{...itemChildProp}
						className={cx('&-Item', itemChildProp.className)}
						onToggle={(isExpanded, { event }) =>
							handleToggle(isExpanded, index, event)
						}
						isExpanded={!itemChildProp.isDisabled && selectedIndex === index}
					/>
				);
			})}
		</div>
	);
};

Accordion.displayName = 'Accordion';

Accordion.propTypes = {
	className: string`
		Appended to the component-specific class names set on the root element.
	`,
	selectedIndex: number`
		Indicates which item is expanded
	`,
	onSelect: func`
		Called when the user clicks on the component's header of an item.
	`,
	style: object`
		Passed through to the root element.
	`,
};

Accordion.peek = {
	description: `\`Accordion\` is a container that renders panels and controls their expansion or collapse.`,
	categories: ['layout'],
	madeFrom: ['ExpanderPanel'],
};

Accordion.defaultProps = defaultProps;

Accordion.reducers = reducers;

Accordion.Item = ExpanderPanel;

Accordion.Header = ExpanderPanel.Header;

export default buildModernHybridComponent<
	IAccordionProps,
	IAccordionState,
	typeof Accordion
>(Accordion, { reducers });

export { Accordion as AccordionDumb };
