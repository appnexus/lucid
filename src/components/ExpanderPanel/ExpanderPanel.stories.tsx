import _ from 'lodash';
import React from 'react';
import createClass from 'create-react-class';
import ExpanderPanel from './ExpanderPanel';

export default {
	title: 'Layout/ExpanderPanel',
	component: ExpanderPanel,
	parameters: {
		docs: {
			description: {
				component: (ExpanderPanel as any).peek.description,
			},
		},
	},
};

/* Basic */
export const Basic = () => {
	const Component = createClass({
		render() {
			return (
				<div>
					<ExpanderPanel>
						<ExpanderPanel.Header>Show More</ExpanderPanel.Header>
						{_.times(100, (n) => (
							<div key={n}>{_.repeat('-', 75 * Math.sin(n / 5))}</div>
						))}
					</ExpanderPanel>
				</div>
			);
		},
	});

	return <Component />;
};

/* No Padding */
export const NoPadding = () => {
	const Component = createClass({
		render() {
			return (
				<div>
					<ExpanderPanel hasPadding={false}>
						<ExpanderPanel.Header>Show More</ExpanderPanel.Header>
						{_.times(100, (n) => (
							<div key={n}>{_.repeat('-', 75 * Math.sin(n / 5))}</div>
						))}
					</ExpanderPanel>
				</div>
			);
		},
	});

	return <Component />;
};
NoPadding.storyName = 'NoPadding';

/* Basic With On Rest Callback */
export const BasicWithOnRestCallback = () => {
	const onRest = () => {
		alert('A big ball of wibbly wobbly, timey wimey stuff');
	};

	const Component = createClass({
		render() {
			return (
				<div>
					<ExpanderPanel onRest={onRest}>
						<ExpanderPanel.Header>Show More</ExpanderPanel.Header>
						{_.times(100, (n) => (
							<div key={n}>{_.repeat('-', 75 * Math.sin(n / 5))}</div>
						))}
					</ExpanderPanel>
				</div>
			);
		},
	});

	return <Component />;
};
BasicWithOnRestCallback.storyName = 'BasicWithOnRestCallback';
