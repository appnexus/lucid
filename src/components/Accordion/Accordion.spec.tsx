import _ from 'lodash';
import assert from 'assert';
import React from 'react';
import sinon from 'sinon';
import { mount, shallow } from 'enzyme';

import { common } from '../../util/generic-tests';
import { AccordionDumb as Accordion } from './Accordion';
import { ExpanderPanelDumb as ExpanderPanel } from '../ExpanderPanel/ExpanderPanel';

describe('Accordion', () => {
	common(Accordion, { exemptChildComponents: ['Header'] });

	describe('props', () => {
		describe('Item', () => {
			it('renders ExpanderPanel components on it', () => {
				const wrapper = mount(
					<Accordion>
						<Accordion.Item Header='Header One'>One</Accordion.Item>
						<Accordion.Item Header='Header Two'>Two</Accordion.Item>
					</Accordion>
				);

				assert(wrapper.find(ExpanderPanel), 2);
			});

			it('Item as children', () => {
				const wrapper = shallow(
					<Accordion>
						<Accordion.Item Header='Header One'>One</Accordion.Item>
						<Accordion.Item Header='Header Two'>Two</Accordion.Item>
					</Accordion>
				);

				assert.equal(wrapper.find('.lucid-Accordion-Item').length, 2);
			});
		});

		describe('Header', () => {
			it('Header as props', () => {
				const wrapper = mount(
					<Accordion>
						<Accordion.Item Header='Froyo'>Yolo fo sho</Accordion.Item>
						<Accordion.Item>Broyoyo</Accordion.Item>
					</Accordion>
				);

				const firstItem = wrapper.find('.lucid-Accordion-Item').first();

				assert.equal(
					firstItem
						.find('.lucid-ExpanderPanel-header')
						.first()
						.text(),
					'Froyo'
				);
			});

			it('Header as children', () => {
				const wrapper = mount(
					<Accordion>
						<Accordion.Item>
							<Accordion.Header>Froyo</Accordion.Header>
							Yolo fo sho
						</Accordion.Item>
						<Accordion.Item>Broyoyo</Accordion.Item>
					</Accordion>
				);

				const firstItem = wrapper.find('.lucid-Accordion-Item').first();

				assert.equal(
					firstItem
						.find('.lucid-ExpanderPanel-header')
						.first()
						.text(),
					'Froyo'
				);
			});
		});

		describe('selectedIndex', () => {
			it('should have an expanded item when set via props', () => {
				const wrapper = mount(
					<Accordion selectedIndex={1}>
						<Accordion.Item Header='Header Test'>test</Accordion.Item>
						<Accordion.Item Header='Header Test'>test</Accordion.Item>
						<Accordion.Item Header='Header Test'>test</Accordion.Item>
					</Accordion>
				);

				assert(wrapper.find('.lucid-ExpanderPanel-content-is-expanded'), 1);
			});
		});

		describe('pass throughs', () => {
			it('passes through all props not defined in `propTypes` to the root element', () => {
				const wrapper = shallow(
					<Accordion
						className='wut'
						style={{ marginRight: 10 }}
						foo={1}
						bar={2}
					/>
				);
				const rootProps = wrapper.find('.lucid-Accordion').props();

				assert(_.has(rootProps, 'foo'), 'props missing "foo" prop');
				assert(_.has(rootProps, 'bar'), 'props missing "bar" prop');
			});

			it('passes through Item className to the rendered item element', () => {
				const wrapper = shallow(
					<Accordion>
						<Accordion.Item className='TestOne'>One</Accordion.Item>
						<Accordion.Item className='TestTwo'>Two</Accordion.Item>
					</Accordion>
				);
				const itemsWrapper = wrapper.find('.lucid-Accordion-Item');

				assert.equal(
					itemsWrapper.find('.TestOne').length,
					1,
					'must find one item with className `TestOne`'
				);
				assert.equal(
					itemsWrapper.find('.TestTwo').length,
					1,
					'must find one item with className `TestTwo`'
				);
			});
		});
	});
});

describe('Accordion', () => {
	let wrapper;
	const onSelect = sinon.spy();
	let mountTestDiv;

	describe('user picks one of the items', () => {
		beforeEach(() => {
			mountTestDiv = document.createElement('div');
			document.body.appendChild(mountTestDiv);
			wrapper = mount(
				<Accordion onSelect={onSelect}>
					<Accordion.Item Header='Header One'>One</Accordion.Item>
					<Accordion.Item Header='Header Two' isDisabled>
						Two
					</Accordion.Item>
				</Accordion>,
				{
					attachTo: mountTestDiv,
				}
			);
		});

		afterEach(() => {
			onSelect.reset();
			wrapper && wrapper.unmount();
			mountTestDiv && mountTestDiv.parentNode.removeChild(mountTestDiv);
		});

		it('should call the function passed in as the `onSelect` prop', () => {
			const firstPanel = wrapper.find('.lucid-ExpanderPanel').at(0);

			firstPanel
				.find('.lucid-ExpanderPanel-header')
				.first()
				.simulate('click');
			firstPanel
				.find('.lucid-ExpanderPanel-icon')
				.first()
				.simulate('click');

			assert.equal(
				onSelect.callCount,
				2,
				`onSelect called the wrong number of times, actual: ${
					onSelect.callCount
				}, expected: 2`
			);
		});

		it('should not call the function passed in as the `onSelect` prop when Item is disabled', () => {
			const secondPanel = wrapper.find('.lucid-ExpanderPanel').at(2);

			secondPanel
				.find('.lucid-ExpanderPanel-header')
				.first()
				.simulate('click');
			secondPanel
				.find('.lucid-ExpanderPanel-icon')
				.first()
				.simulate('click');

			expect(onSelect.callCount).toEqual(0);
		});
	});
});
