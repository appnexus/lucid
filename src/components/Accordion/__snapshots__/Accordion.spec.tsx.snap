// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion [common] example testing should match snapshot(s) for 1.basic 1`] = `
<div
  className="lucid-Accordion"
>
  <ExpanderPanel
    Header="Peter Venkman"
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={false}
    isExpanded={false}
    key="0"
    onToggle={[Function]}
  >
    Peter Venkman, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters and Ghostbusters II and in the animated television series The Real Ghostbusters. In both live action films, he was portrayed by Bill Murray, and was voiced in the animated series first by Lorenzo Music and then by Dave Coulier. He is a parapsychologist and the leader of the Ghostbusters.
  </ExpanderPanel>
  <ExpanderPanel
    Header="Ray Stantz"
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={false}
    isExpanded={false}
    key="1"
    onToggle={[Function]}
  >
    Raymond Ray Stantz, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters, Ghostbusters II, Casper, and the animated television series The Real Ghostbusters. He was portrayed by Dan Aykroyd in both live action films, and voiced by Frank Welker in the animated series. He is a member of the Ghostbusters and one of the three doctors of parapsychology, along with Dr. Peter Venkman and Dr. Egon Spengler.
  </ExpanderPanel>
  <ExpanderPanel
    Header="Egon Spengler"
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={false}
    isExpanded={false}
    key="2"
    onToggle={[Function]}
  >
    Egon Spengler, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters and Ghostbusters II, in the animated television series The Real Ghostbusters, and later in Extreme Ghostbusters. Spengler was portrayed by Harold Ramis in the films and voiced by him in Ghostbusters: The Video Game, and voiced by Maurice LaMarche in the cartoon series. He is a member of the Ghostbusters and one of the three doctors of parapsychology, along with Dr. Peter Venkman and Dr. Raymond Stantz.
  </ExpanderPanel>
</div>
`;

exports[`Accordion [common] example testing should match snapshot(s) for 2.initial-index-set 1`] = `
<div
  className="lucid-Accordion"
>
  <ExpanderPanel
    Header="Peter Venkman"
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={false}
    isExpanded={false}
    key="0"
    onToggle={[Function]}
  >
    Peter Venkman, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters and Ghostbusters II and in the animated television series The Real Ghostbusters. In both live action films, he was portrayed by Bill Murray, and was voiced in the animated series first by Lorenzo Music and then by Dave Coulier. He is a parapsychologist and the leader of the Ghostbusters.
  </ExpanderPanel>
  <ExpanderPanel
    Header="Ray Stantz"
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={false}
    isExpanded={false}
    key="1"
    onToggle={[Function]}
  >
    Raymond Ray Stantz, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters, Ghostbusters II, Casper, and the animated television series The Real Ghostbusters. He was portrayed by Dan Aykroyd in both live action films, and voiced by Frank Welker in the animated series. He is a member of the Ghostbusters and one of the three doctors of parapsychology, along with Dr. Peter Venkman and Dr. Egon Spengler.
  </ExpanderPanel>
  <ExpanderPanel
    Header="Egon Spengler"
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={false}
    isExpanded={true}
    key="2"
    onToggle={[Function]}
  >
    Egon Spengler, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters and Ghostbusters II, in the animated television series The Real Ghostbusters, and later in Extreme Ghostbusters. Spengler was portrayed by Harold Ramis in the films and voiced by him in Ghostbusters: The Video Game, and voiced by Maurice LaMarche in the cartoon series. He is a member of the Ghostbusters and one of the three doctors of parapsychology, along with Dr. Peter Venkman and Dr. Raymond Stantz.
  </ExpanderPanel>
</div>
`;

exports[`Accordion [common] example testing should match snapshot(s) for 3.fixed-index 1`] = `
<div
  className="lucid-Accordion"
>
  <ExpanderPanel
    Header="Peter Venkman"
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={false}
    isExpanded={false}
    key="0"
    onToggle={[Function]}
  >
    Peter Venkman, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters and Ghostbusters II and in the animated television series The Real Ghostbusters. In both live action films, he was portrayed by Bill Murray, and was voiced in the animated series first by Lorenzo Music and then by Dave Coulier. He is a parapsychologist and the leader of the Ghostbusters.
  </ExpanderPanel>
  <ExpanderPanel
    Header="Ray Stantz"
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={false}
    isExpanded={true}
    key="1"
    onToggle={[Function]}
  >
    Raymond Ray Stantz, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters, Ghostbusters II, Casper, and the animated television series The Real Ghostbusters. He was portrayed by Dan Aykroyd in both live action films, and voiced by Frank Welker in the animated series. He is a member of the Ghostbusters and one of the three doctors of parapsychology, along with Dr. Peter Venkman and Dr. Egon Spengler.
  </ExpanderPanel>
  <ExpanderPanel
    Header="Egon Spengler"
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={false}
    isExpanded={false}
    key="2"
    onToggle={[Function]}
  >
    Egon Spengler, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters and Ghostbusters II, in the animated television series The Real Ghostbusters, and later in Extreme Ghostbusters. Spengler was portrayed by Harold Ramis in the films and voiced by him in Ghostbusters: The Video Game, and voiced by Maurice LaMarche in the cartoon series. He is a member of the Ghostbusters and one of the three doctors of parapsychology, along with Dr. Peter Venkman and Dr. Raymond Stantz.
  </ExpanderPanel>
</div>
`;

exports[`Accordion [common] example testing should match snapshot(s) for 4.disabled-item 1`] = `
<div
  className="lucid-Accordion"
>
  <ExpanderPanel
    Header="Peter Venkman"
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={false}
    isExpanded={false}
    key="0"
    onToggle={[Function]}
  >
    Peter Venkman, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters and Ghostbusters II and in the animated television series The Real Ghostbusters. In both live action films, he was portrayed by Bill Murray, and was voiced in the animated series first by Lorenzo Music and then by Dave Coulier. He is a parapsychologist and the leader of the Ghostbusters.
  </ExpanderPanel>
  <ExpanderPanel
    Header="Ray Stantz"
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={true}
    isExpanded={false}
    key="1"
    onToggle={[Function]}
  >
    Raymond Ray Stantz, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters, Ghostbusters II, Casper, and the animated television series The Real Ghostbusters. He was portrayed by Dan Aykroyd in both live action films, and voiced by Frank Welker in the animated series. He is a member of the Ghostbusters and one of the three doctors of parapsychology, along with Dr. Peter Venkman and Dr. Egon Spengler.
  </ExpanderPanel>
  <ExpanderPanel
    Header="Egon Spengler"
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={false}
    isExpanded={false}
    key="2"
    onToggle={[Function]}
  >
    Egon Spengler, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters and Ghostbusters II, in the animated television series The Real Ghostbusters, and later in Extreme Ghostbusters. Spengler was portrayed by Harold Ramis in the films and voiced by him in Ghostbusters: The Video Game, and voiced by Maurice LaMarche in the cartoon series. He is a member of the Ghostbusters and one of the three doctors of parapsychology, along with Dr. Peter Venkman and Dr. Raymond Stantz.
  </ExpanderPanel>
</div>
`;

exports[`Accordion [common] example testing should match snapshot(s) for 5.using-header-as-child 1`] = `
<div
  className="lucid-Accordion"
>
  <ExpanderPanel
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={false}
    isExpanded={false}
    key="0"
    onToggle={[Function]}
  >
    <ExpanderPanel.Header>
      Peter Venkman
    </ExpanderPanel.Header>
    Peter Venkman, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters and Ghostbusters II and in the animated television series The Real Ghostbusters. In both live action films, he was portrayed by Bill Murray, and was voiced in the animated series first by Lorenzo Music and then by Dave Coulier. He is a parapsychologist and the leader of the Ghostbusters.
  </ExpanderPanel>
  <ExpanderPanel
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={true}
    isExpanded={false}
    key="1"
    onToggle={[Function]}
  >
    <ExpanderPanel.Header>
      Ray Stantz
    </ExpanderPanel.Header>
    Raymond Ray Stantz, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters, Ghostbusters II, Casper, and the animated television series The Real Ghostbusters. He was portrayed by Dan Aykroyd in both live action films, and voiced by Frank Welker in the animated series. He is a member of the Ghostbusters and one of the three doctors of parapsychology, along with Dr. Peter Venkman and Dr. Egon Spengler.
  </ExpanderPanel>
  <ExpanderPanel
    className="lucid-Accordion-Item"
    hasPadding={true}
    isDisabled={false}
    isExpanded={false}
    key="2"
    onToggle={[Function]}
  >
    <ExpanderPanel.Header>
      Egon Spengler
    </ExpanderPanel.Header>
    Egon Spengler, Ph.D. is a fictional character from the Ghostbusters franchise. He appears in the films Ghostbusters and Ghostbusters II, in the animated television series The Real Ghostbusters, and later in Extreme Ghostbusters. Spengler was portrayed by Harold Ramis in the films and voiced by him in Ghostbusters: The Video Game, and voiced by Maurice LaMarche in the cartoon series. He is a member of the Ghostbusters and one of the three doctors of parapsychology, along with Dr. Peter Venkman and Dr. Raymond Stantz.
  </ExpanderPanel>
</div>
`;
